packages/core/src/root.ts

```diff
@@ -117,8 +117,22 @@ export function createRemoteRoot<
       }
 
       if (initialChildren) {
-        for (const child of initialChildren) {
-          normalizedInitialChildren.push(normalizeChild(child, remoteRoot));
+        if (Array.isArray(initialChildren)) {
+          for (const child of initialChildren) {
+            normalizedInitialChildren.push(normalizeChild(child, remoteRoot));
+          }
+        } else {
+          normalizedInitialChildren.push(
+            normalizeChild(initialChildren, remoteRoot),
+          );
+
+          // The complex tuple type of `rest` makes it so `moreChildren` is
+          // incorrectly inferred as potentially being the props of the component,
+          // lazy casting since we know it will be an array of child elements
+          // (or empty).
+          for (const child of moreChildren as any[]) {
+            normalizedInitialChildren.push(normalizeChild(child, remoteRoot));
```

Comment:
I think it's possible that `undefined` gets pushed onto `normalizedInitialChildren` when `initialChild`, but no `moreChildren` are passed.

Is this being sanitized later or could it lead to issues?

Reply:
True, I got that confused. When you omit `moreChildren` completely it's an empty array. 👍